protos/sigstore_trustroot.proto

```diff
@@ -0,0 +1,86 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.trustroot.v1;
+
+import "sigstore_common.proto";
+
+option go_package = "github.com/sigstore/proto/trustroot/v1";
+option java_package = "dev.sigstore.proto.trustroot.v1";
+option java_multiple_files = true;
+option java_outer_classname = "TrustRootProto";
+
+// TransparencyLogInstance describes the immutable parameters from a
+// transparency log.
+// See https://www.rfc-editor.org/rfc/rfc9162.html#name-log-parameters
+// for more details.
+// The incluced parameters are the minimal set required to identify a log,
+// and verify an inclusion promise.
+message TransparencyLogInstance {
+        // The base URL at which can be used to URLs for the client.
+        string base_url = 1;
+        // The hash algorithm used for the Merkle Tree.
+        dev.sigstore.common.v1.HashAlgorithm hash_algorithm = 2;
+        // The public key used to verify signatures generated by the log.
+        // This attribute contains the signature algorithm used by the log.
+        dev.sigstore.common.v1.PublicKey public_key = 3;
+        // The unique identifier for this transparency log.
+        dev.sigstore.common.v1.LogId log_id = 4;
+}
+
+// CertificateAuthority enlists the information required to identify which
+// CA to use and perform signature verification.
+message CertificateAuthority {
+        // The root certificate MUST be self-signed, and so the subject and
+        // issuer are the same.
+        dev.sigstore.common.v1.DistinguishedName subject = 1;
+        // The URI at which the CA can be accessed.
+        string uri = 2;
+        // The certificate chain for this CA.
+        dev.sigstore.common.v1.X509CertificateChain cert_chain = 3;
+        // The time the *entire* chain was valid. This is at max the
+        // longest interval when *all* certificates in the chain where valid,
+        // but it MAY be shorter.
+        dev.sigstore.common.v1.TimeRange valid_for = 4;
+}
+
+// TrustedRoot describes the client's complete set of trusted entities.
+// How the TrustedRoot is populated is not specified, but can be a
+// combination of many sources such as TUF repositories, files on disk etc.
+//
+// The TrustedRoot is not meant to be used for any artifact verification, only
+// to capture the complete/global set of trusted verification materials.
+// When verifying an artifact, based on the artifact and policies, a selection
```

Comment:
Just thinking out loud if CA & TransparencyInstance has enough information in there to write policies against. Both of them have a URI / base_url and CA has DistinguishedName, but just curious if that's enough and that's what one would use to filter out the instances that should be used for verification.

Reply:
I guess thinking that the one (only?) reason for having the TrustedRoot as a collection of all the everythings is that you can then pass it through some sort of policy machine which will distill it down to the verification materials before calling the actual verification with a subset of keys/authorities.

Reply:
> I guess thinking that the one (only?) reason for having the TrustedRoot as a collection of all the everythings is that you can then pass it through some sort of policy machine which will distill it down to the verification materials before calling the actual verification with a subset of keys/authorities.

Yes, that's the exact reasoning. 

Based on the scenarios I'm aware of, we should have enough data to confidently distill what exact "instance" of a CA or log to use:
* The name (log_id or distinguished name)
* URI
* Time range the service was active for

What could possibly cause some confusion is that if a client is relying on a TUF multi root setup, where multiple TUF roots exposes targets for the log or CA with identical data. But this is an explicit confusion as I see it, and so the precondition that the client trusts the TUF root may not hold any more, then all bets are off, as the TUF repository may serve arbitrary data. But I don't think of this as a real problem.

What the policy will protect against is e.g. where an artifact was signed with a certificate from the wrong CA given a policy, and for that I'd say we are good.

Reply:
Fantastic. Again just thinking of how the layer above this (policy-controller) will sort out the details. But, I guess that's my problem :)

Reply:
I'm working a bit on that too, I hope that I can share some details on this later today!